### DISCUSSION TITLE
How to visualize the point cloud published by a realsense device in YARP?

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/community#684] [Q&A] [answered]
I want to visualize the point cloud publised by a realsense device via https://github.com/robotology/yarp-device-realsense2 . For the RGB feed I can use `yarpview`, anyone knows what is the easiest way to visualize the point cloud? Perhaps via rviz with a specific ROS nws? 

Thanks a lot in advance!

fyi @Nicogene

**Comment by traversaro** [comment]
cc @Gio-DS

**Reply by pattacini** [reply]
cc @randaz81 @xEnVrE

**Comment by randaz81** [comment, marked as answer]
Realsense gives you rgb and depth images. Before visualizing something on rviz, you have to compute a pointcloud. You can use: https://github.com/robotology/yarp-devices-ros2/tree/master/src%2Fdevices%2FrgbdToPointCloudSensor_nws_ros2.
Since poincloud is an expensive datatype in terms of used memory and bandwidth, it is highly recommended to transmit over the network compressed rgb and depth images, and create the pointcloud locally, as the final step of the pipeline.
